File: python/src/problems1/find_the_index_of_the_first_occurrence_in_a_string.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def str_str(self, haystack: str, needle: str) -> int:
        pass


class Default(Problem):
    @override
    def str_str(self, haystack: str, needle: str) -> int:
        return haystack.find(needle)


SOLUTIONS: list[Problem] = [Default()]

File: javascript/src/problems1_100/find-the-index-of-the-first-occurrence-in-a-string.js
class Problem {
  strStr(haystack, needle) {
    throw new Error(`Not implemented (${haystack}, ${needle}).`);
  }
}

class Default extends Problem {
  strStr(haystack, needle) {
    return haystack.indexOf(needle);
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;
